.github/workflows/e2e-nvidia-a10g-x1.yml

```diff
@@ -116,18 +117,41 @@ jobs:
 
       - name: Install ilab
         run: |
-          export PATH="/home/ec2-user/.local/bin:/usr/local/cuda/bin:$PATH"
-          python3.11 -m venv --upgrade-deps venv
+          export CUDA_HOME="/usr/local/cuda"
+          export LD_LIBRRY_PATH="$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64"
+          export PATH="$PATH:$CUDA_HOME/bin"
+          python3.11 -m venv venv
           . venv/bin/activate
+          nvidia-smi
           sed 's/\[.*\]//' requirements.txt > constraints.txt
           python3.11 -m pip cache remove llama_cpp_python
           CMAKE_ARGS="-DLLAMA_CUBLAS=on" python3.11 -m pip install --force-reinstall --no-binary llama_cpp_python -c constraints.txt llama_cpp_python
           python3.11 -m pip install bitsandbytes
+
+          # TODO This should be added to instructlab-training
+          python3.11 -m pip install packaging wheel
+
+          python3.11 -m pip install instructlab-training[cuda]
+
           python3.11 -m pip install .
 
       - name: Run e2e test
         run: |
+          nvidia-smi
+          export PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128
```

Comment:
Do you know why this was needed? Can you add a comment explaining what's going on?  Also, is this something that needs to go in our docs?

Reply:
PYTORCH_CUDA_ALLOC_CONF makes the chunks cuda can alloc at once smaller, helps avoid OOM. I don't think this needs to be documented on our end

Reply:
if we needed it, then it must be useful to others

Reply:
If we're changing the default cuda operations for a test I personally think we should note the rationale. Prevents tribal knowledge

Reply:
and the explanation should include why we need it here, but why it's not appropriate to be the default behavior

and should include docs for others to know when they made to set it to achieve the same success using `ilab`